### DISCUSSION TITLE
my image on the right side won't align with the left text

### DISCUSSION THREAD

**Discussion by Selyiah** [community/community#78708] [Programming Help] [answered]
### Body

<img width="1282" alt="Screenshot 2023-12-11 at 21 35 47" src="https://github.com/community/community/assets/140277594/ec9b41a6-5cc5-4f9b-90a7-20263e906a0e">

The HTML: 

```
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Group Project</title>
    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.2/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-T3c6CoIi6uLrA9TneNEoa7RxnatzjcDSCmG1MXxSR1GAsXEV/Dwwykc2MPK8M2HN" crossorigin="anonymous">
    <!-- this link above has been taken from getbootstrap.com via CDN download, copied the top link -->
    <link rel="stylesheet" href="homepage.css"/>
</head>
<body>
<div class="card">
   <div class="row">
      <div class="float-start">
        <div class="card-body">
          <h1 class="card-title">Discover London's Best: <br> Choose, Rate and Experience </br><br1> Your Favourites</br1></h1>
          <p class="card-text">With city recommendations from other hotspotters who <br> have tried and tested! </br></p>
        <div class="d-grid gap-2 d-md-block">
          <button class="btn btn-dark" type="button">Get Started</button>
          <button class="btn btn-light" type="button">Learn More</button>
        <div class="float-end">
          <img class="img-responsive" src="feed-page.png" alt="">
        </div>
      </div>  
   </div>
 </div>
```

The CSS: 

```
.card {
    align-items: center;
    justify-content: center;
    display: flex;
}    
   

.img-responsive {
    margin-right: 90px;
    max-width: 450px;
    max-height: 450px;
}

.card-body {
    margin-top: 250px;
    margin-left: 90px;
}

body {
    margin: 50px;
}
```

Please help me see why it's not aligning next to each other, thank you community! 

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by seyLu** [comment, marked as answer]
#### Hi @Selyiah,

Whenever you're debugging in the browser - in this case, the image alignment - it is a good practice to use the built-in inspect tool found in most browsers:

<kbd><kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>I</kbd></kbd>
or right-click + inspect
 
![image](https://github.com/community/community/assets/98249191/af98fef3-1490-4995-952d-5978f4e21e46)

Click the select tool:

![image](https://github.com/community/community/assets/98249191/209d77a0-251d-406f-9b5a-a5790d0c903d)

And start exploring:
(see what happens when you remove a class, or when you toggle styles, etc.)

![image](https://github.com/community/community/assets/98249191/27832fa5-43d1-4135-ac57-9c3eb28a5390)

---

<br>

In this case, you are using display flex with center alignment on card class, but at the same time, using bootstrap's row class on the child element

<table><tbody><tr>
<td width="300px">

```css
.card {
  align-items: center;
  justify-content: center;
  display: flex;
}
```

</td>
<td width="300px">

```html
<div class="card">
   <div class="row">
      ...
    </div>
</div>
```
</tr></tbody><table>

Flex alignment only works on direct descendants. Thus, if you remove child elems that are not needed, you will see the flex alignment take effect:

```html
<div class="card">
   <div class="card-body">
     ... 
   </div>
   <img src="https://picsum.photos/400/500" alt="image">
</div>
```
![image](https://github.com/community/community/assets/98249191/40658613-7e52-4857-8be5-7fa64f5a4b78)

And if you've done your homework and studied how flexbox and grids work, then it would be intuitive to think to just change the flex direction and that should fix the issue:

```css
.card {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;

  & .card-body {
    padding: 2rem;
  }
}
```

![image](https://github.com/community/community/assets/98249191/028358d9-e2ba-44b6-923a-440d93f382b1)

**Reply by Selyiah** [reply]
Thank you so much! Very helpful
